### ISSUE TITLE
Jpackage out or err stream encoding not correct.

### ISSUE THREAD

**Issue by iblader** [wiverson/jtoolprovider-plugin#19]
RunTool should not enforce using utf-8 encoding for the out or err PrintStream, but should use platform default encoding, or the output of jpackage messages tool will become '?????' when it is localized. 

PS: JDK 17 on windows 11, chinese language, the default OS encoding is GBK.

**Comment by wiverson** [comment]
Hmm. Went with utf-8 to a) keep simple and b) try to encourage folks to get get away from plat specific encodings.

Want to submit a PR to add as an option? FWIW I can take a look at adding a look at the best way to do this (pick up project encoding as default with optional override? Set to utf-8 by default w/optional override?) Probably won't be able to get to this until Dec...

**Comment by iblader** [comment]
I think the simplest solution is to change the PrintStream constructor to this one:
` new PrintStream(outputStream, true)`
without appoint the encoding parameter, just let the jvm to decide which encode should be used.

**Comment by wiverson** [comment]
Yeah, Maven has a place you are supposed to set the encoding as a value e.g. https://maven.apache.org/plugins/maven-resources-plugin/examples/encoding.html

And so I think off the cuff it's os/environment, the jvm encoding, and the maven build encoding that are all plausible. Plus of course the JDK [moved to UTF-8 by default](https://openjdk.org/jeps/400) in JDK 18.

I *think* the best option is to add an encoding property for the plugin that defaults to the project.build.sourceEncoding encoding. So if you did nothing it would use the same encoding as the project but you could also override per execution.

**Comment by iblader** [comment]
Yes, I agree the most flex way is to add a encoding property to the out and err PrintStream. 
The maven resource plugin encoding is another story, I met this "windows-jpackage.txt" encoding problem as well, I found the jpackage tool read config file by default encoding, but the maven resource plugin filter this config by uft-8, I solved it by config this plugin in windows build profile:
```
<plugins>
    <plugin>
         <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-resources-plugin</artifactId>
            <version>3.3.0</version>
               <configuration>
                    <encoding>GBK</encoding>
              </configuration>
   </plugin>
</plugins>
```
